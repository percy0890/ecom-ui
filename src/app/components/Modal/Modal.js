import PropTypes from 'prop-types';
import React from 'react';
import './Modal.scss';

export default function Modal(props) {
  const {
    isModalOpen,
    children,
    showCloseIcon = true,
    closeOnOuterClick = false,
    closeModal,
    customClass,
  } = props;
  return (
    isModalOpen && (
      <div
        className="modal"
        onClick={closeOnOuterClick ? closeModal : () => {}}
        role="presentation"
      >
        <div className={`modal-content app-container ${customClass}`}>
          {children}
          {showCloseIcon && (
            <button
              type="button"
              tabIndex={0}
              className="close"
              onClick={closeModal}
              onKeyUp={closeModal}
            >
              &times;
            </button>
          )}
        </div>
      </div>
    )
  );
}

Modal.propTypes = {
  isModalOpen: PropTypes.bool.isRequired,
  children: PropTypes.any,
  closeModal: PropTypes.func.isRequired,
  showCloseIcon: PropTypes.bool,
  customClass: PropTypes.string,
  closeOnOuterClick: PropTypes.bool,
};
