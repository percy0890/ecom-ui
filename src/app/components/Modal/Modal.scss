@import '~app/styles/core';

/* The Modal (background) */
.modal {
  position: fixed;
  /* Stay in place */
  z-index: 1000;
  /* Sit on top */
  /* Location of the box */
  left: 0;
  top: 0;
  width: 100%;
  /* Full width */
  height: 100%;
  /* Full height */
  overflow: auto;
  /* Enable scroll if needed */
  background-color: rgb(0,0,0);
  /* Fallback color */
  background-color: rgba(0,0,0,0.6);
  /* Black w/ opacity */

  /* The Close Button */
  .close {
    position: absolute;
    top: 0rem;
    right: 0rem;
    font-size: 2.4rem;
    color: $app-text-primary;
    border: 0;
    cursor: pointer;
  }

  /* .close:focus,
  .close:hover {
    color: #fff;
    background: #222;
    border-color: #222;
    text-decoration: none;
  } */
}

/* Modal Content */
.modal-content {
  background-color: #fff;
  margin: auto;
  padding: 15px;
  border: 1px solid #666;
  border-radius: 4px;
  box-shadow: 0 0 30px #666;
  color: #222;
  width: 80vw;
  position: relative;
}
