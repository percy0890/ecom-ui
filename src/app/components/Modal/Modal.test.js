import React from 'react';
import { shallow } from 'enzyme';
import Modal from '.';

describe('Modal Component', () => {
  let wrapper;

  it('renders correctly', () => {
    wrapper = shallow(
      <Modal isModalOpen closeModal={jest.fn()} closeOnOuterClick />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with outer-click disabled', () => {
    wrapper = shallow(<Modal isModalOpen closeModal={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
