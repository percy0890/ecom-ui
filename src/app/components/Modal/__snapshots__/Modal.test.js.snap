// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders correctly 1`] = `
<div
  className="modal"
  onClick={[MockFunction]}
  role="presentation"
>
  <div
    className="modal-content app-container undefined"
  >
    <button
      className="close"
      onClick={[MockFunction]}
      onKeyUp={[MockFunction]}
      tabIndex={0}
      type="button"
    >
      ×
    </button>
  </div>
</div>
`;

exports[`Modal Component renders correctly with outer-click disabled 1`] = `
<div
  className="modal"
  onClick={[Function]}
  role="presentation"
>
  <div
    className="modal-content app-container undefined"
  >
    <button
      className="close"
      onClick={[MockFunction]}
      onKeyUp={[MockFunction]}
      tabIndex={0}
      type="button"
    >
      ×
    </button>
  </div>
</div>
`;
